spec/models/merchant_spec.rb

```diff
@@ -1,20 +1,52 @@
 require 'rails_helper'
 
-RSpec.describe Merchant do
+RSpec.describe Merchant, type: :model do
+  before :each do
+    setup_data
+  end
+
   describe 'relationhips' do
     it { should have_many :items }
     it { should have_many(:invoice_items).through(:items)}
   end
 
-  before :each do
+  describe 'validations' do
+    it { should define_enum_for(:status).with_values(disabled: 0, enabled: 1) }
+  end
+
+  describe 'instance methods' do
+    it 'finds the top five customers' do
+      expected = [@customer_1, @customer_5, @customer_4, @customer_3, @customer_2]
+
+      expect(@merchant.top_five_customers).to eq(expected)
+    end
+
+    it 'finds transaction count given a customer_id' do
+      expect(@merchant.transaction_count(@customer_1.id)).to eq(5)
+    end
+
+    it 'returns items for the merchant that need to be shipped' do
+      expect(@merchant.invoice_items_ready).to eq([@invoice_item_1])
+      expect(@merchant.invoice_items_ready).not_to include(@invoice_item_2)
+    end
+
+    it 'returns the invoice date for an item ready to ship' do
+      expect(@merchant.item_invoice_date(@invoice_1.id)).to eq(sample_date.strftime('%A, %B %d, %Y'))
+    end
+  end
+
+  def sample_date
+    DateTime.new(2021, 01, 01)
+  end
+
+  def setup_data
```

Comment:
THIS IS SUCH A GOOD IDEA! My feature tests where getting unwieldy in the setups, even using factorybot! I am going to refactor my test code to follow this organization.

Reply:
Yeah, I didn't want to scroll through all of the setup code there just to get to the tests, so that helper method could be put at the bottom of the file so as to not get in the way.  IMPORTANT NOTE: if you add helper methods make sure they're _inside_ the outermost `describe` block. If a method is outside of that outermost describe block, they end up being in a global scope where other tests can access them (and that will cause problems if helper methods in different tests share the same name...)